src/utils/warnTooManyClasses.js

```diff
@@ -0,0 +1,27 @@
+import styleSheet from '../models/StyleSheet'
+
+const components = WeakMap ? new WeakMap() : undefined
```

Comment:
This code will run only in development mode. I assumed usage of the modern browser in development. If WeakMap is not supported - no checks will be made.

It was the easiest way to keep changes in the main codebase minimal, but I'll take a look at removing it.